command/login/login.go

```diff
@@ -0,0 +1,148 @@
+package login
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/command/flags"
+	"github.com/hashicorp/consul/lib/file"
+	"github.com/mitchellh/cli"
+)
+
+func New(ui cli.Ui) *cmd {
+	c := &cmd{UI: ui}
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+	http  *flags.HTTPFlags
+	help  string
+
+	shutdownCh <-chan struct{}
+
+	bearerToken string
+
+	// flags
+	authMethodName  string
+	bearerTokenFile string
+	tokenSinkFile   string
+	meta            map[string]string
+}
+
+func (c *cmd) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	c.flags.StringVar(&c.authMethodName, "method", "",
+		"Name of the auth method to login to.")
+
+	c.flags.StringVar(&c.bearerTokenFile, "bearer-token-file", "",
```

Comment:
Is there any reason this has to be a file? Why not use the '@' prefix to signify its from a file?

Reply:
Wanted to discourage from incidentally putting the bearer token into the process listing on the server.

Reply:
Seems reasonable